terraform/functions.tf

```diff
@@ -134,7 +134,7 @@ module "reporting_data_bucket" {
   cors_configuration = [
     {
       allowed_methods = ["GET", "HEAD", "POST", "PUT"]
```

Comment:
should we add `OPTIONS` as well @TylerHendrickson ?

Reply:
Generally no, `OPTIONS` shouldn't be necessary (well-noted that our API Gateway configuration does specify `OPTIONS`) for CORS configuration.

[This StackOverflow answer](https://stackoverflow.com/a/68529748) goes a bit deeper into the reason why, but I think the short version is just that a server doesn't need to explicitly allow `OPTIONS` requests because that's the mechanism by which allowed methods are determined in the first place. Including `OPTIONS` assumes that a browser would make a preflight (`OPTIONS` method) request with an `Access-Control-Request-Method: OPTIONS` header, which is a bit like calling someone on the phone to ask if you are allowed to call them on the phone.